lib/mbaasRequest/mbaasRequest.js

```diff
@@ -177,15 +177,20 @@ function responseHandler(params, cb) {
     if (err || (httpResponse && httpResponse.statusCode >= 400)) {
 
       //There is a specific error message if an environment is unreachable
-      if (httpResponse && httpResponse.statusCode) {
+      if (httpResponse && (httpResponse.statusCode || err.code)) {
         err = err || {};
-        err.httpCode = httpResponse.statusCode;
+        err.httpCode = httpResponse.statusCode || err.code;
```

Comment:
`err.httpCode` is only for http Codes so I don't think `|| err.code` is required.